src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPolymorphicTypeConfiguration.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.Json.Serialization.Metadata;
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Defines polymorphic configuration for a specified base type.
+    /// </summary>
+    [System.Runtime.Versioning.RequiresPreviewFeatures("Polymorphic serialization is currently in preview.")]
+    public class JsonPolymorphicTypeConfiguration : IJsonPolymorphicTypeConfiguration, ICollection<(Type DerivedType, string? TypeDiscriminatorId)>
+    {
+        private readonly List<(Type DerivedType, string? TypeDiscriminatorId)> _derivedTypes = new();
+        private string? _customTypeDiscriminatorPropertyName;
+        private JsonUnknownDerivedTypeHandling _unknownDerivedTypeHandling;
+        private bool _ignoreUnrecognizedTypeDiscriminators;
+
+        /// <summary>
+        /// Creates a new polymorphic configuration instance for a given base type.
+        /// </summary>
+        /// <param name="baseType">The base type for which to configure polymorphic serialization.</param>
+        public JsonPolymorphicTypeConfiguration(Type baseType)
+        {
+            if (!PolymorphicTypeResolver.IsSupportedPolymorphicBaseType(baseType))
+            {
+                throw new ArgumentException(SR.Format(SR.Polymorphism_TypeDoesNotSupportPolymorphism, baseType), nameof(baseType));
+            }
+
+            BaseType = baseType;
+        }
+
+        /// <summary>
+        /// Gets the base type for which polymorphic serialization is being configured.
+        /// </summary>
+        public Type BaseType { get; }
+
+        /// <summary>
+        /// Gets or sets the behavior when serializing an undeclared derived runtime type.
+        /// </summary>
+        public JsonUnknownDerivedTypeHandling UnknownDerivedTypeHandling
+        {
+            get => _unknownDerivedTypeHandling;
+            set
+            {
+                VerifyMutable();
+                _unknownDerivedTypeHandling = value;
+            }
+        }
+
+        /// <summary>
+        /// When set to <see langword="true"/>, instructs the deserializer to ignore any
```

Comment:
nit: we usually reference "deserialization" rather than refer to `JsonSerializer` as "the deserializer".

Reply:
Fixed.